test/src/androidTest/kotlin/io/realm/shared/RealmListTests.kt

```diff
@@ -806,6 +827,91 @@ internal abstract class ManagedListTester<T>(
         }
     }
 
+    override fun observeEmitListOnListCollect() {
+        val dataSet = typeSafetyManager.getInitialDataSet()
+
+        errorCatcher {
+            realm.writeBlocking {
+                val list = typeSafetyManager.createContainerAndGetList(this)
+                list.addAll(copyToRealmIfNeeded(dataSet))
+            }
+
+            runBlocking {
+                val channel = Channel<RealmList<*>>(capacity = 1)
+                val observer = async {
+                    val container = realm.objects<RealmListContainer>()
+                        .first()
+
+                    typeSafetyManager.getList(container)
+                        .observe()
+                        .onEach { flowList ->
+                            channel.send(flowList)
+                        }.collect()
+                }
+
+                // Assertion after empty list is emitted
+                val emittedList = channel.receive()
+                assertNotNull(emittedList)
+                assertEquals(dataSet.size, emittedList.size)
+
+                observer.cancel()
+                channel.close()
+            }
+        }
+        assertListAndCleanup { /* No additional assertions */ }
+    }
+
+    override fun observeEmitListOnListUpdates() {
+        val dataSet = typeSafetyManager.getInitialDataSet()
+
+        errorCatcher {
+            realm.writeBlocking {
+                // Just create an empty container with empty lists
+                typeSafetyManager.createContainerAndGetList(this)
+            }
+
+            runBlocking {
+                val channel = Channel<RealmList<*>>(capacity = 2)
+                val observer = async {
+                    val container = realm.objects<RealmListContainer>()
+                        .first()
+
+                    typeSafetyManager.getList(container)
+                        .observe()
+                        .onEach { flowList ->
+                            println("--- ONEACH SIZE: ${flowList.size}")
+
+                            channel.send(flowList)
+
+                            // Update list if the flow list is empty
+                            if (flowList.isEmpty()) {
```

Comment:
Why is this block in here. Couldn't you just do the updates after you have asserted on the empty list below. The idea with the channel-pattern is that you can write actions and assertions sequentially.